attribute_set/models/attribute_attribute.py

```diff
@@ -0,0 +1,480 @@
+                    [("attribute_id", "=", self.id)]
+                ).unlink()
+
+    def _delete_old_fields_options(self, options):
+        """Delete attribute's field values in the objects using our attribute
+        as a field, if these values are not in the new Domain or Options list
+        """
+        self.ensure_one()
+        custom_field = self.name
+        for obj in self.env[self.model].search([]):
+            if obj.fields_get(custom_field):
+                for value in obj[custom_field]:
+                    if value not in options:
+                        if self.attribute_type == "select":
+                            obj.write({custom_field: False})
+                        elif self.attribute_type == "multiselect":
+                            obj.write({custom_field: [(3, value.id, 0)]})
+
+    def write(self, vals):
+        # Prevent from changing Attribute's type
+        if "attribute_type" in list(vals.keys()):
+            if self.search(
+                [
+                    ("attribute_type", "!=", vals["attribute_type"]),
+                    ("id", "in", self.ids),
+                ]
+            ):
+                raise ValidationError(
+                    _(
+                        "Can't change the type of an attribute. "
+                        "Please create a new one."
+                    )
+                )
+            else:
+                vals.pop("attribute_type")
+        # Prevent from changing relation_model_id for multiselect Attributes
+        # as the values of the existing many2many Attribute fields won't be
+        # deleted if changing relation_model_id
+        if "relation_model_id" in list(vals.keys()):
+            if self.search(
+                [
+                    ("relation_model_id", "!=", vals["relation_model_id"]),
+                    ("id", "in", self.ids),
+                ]
+            ):
+                raise ValidationError(
+                    _(
+                        """Can't change the attribute's Relational Model in order to
+                        avoid conflicts with existing objects using this attribute.
+                        Please create a new one."""
+                    )
+                )
+        # Prevent from changing 'Serialized'
+        if "serialized" in list(vals.keys()):
+            if self.search(
+                [("serialized", "!=", vals["serialized"]), ("id", "in", self.ids)]
+            ):
+                raise ValidationError(
+                    _(
+                        """It is not allowed to change the boolean 'Serialized'.
+                        A serialized field can not be change to non-serialized \
+                        and vice versa."""
+                    )
+                )
+        # Set the new values to self
+        res = super().write(vals)
+
+        for att in self:
+            options = att.option_ids
+            if self.relation_model_id:
```

Comment:
This is broken here, no?
```suggestion
            if att.relation_model_id:
```